### PR TITLE
[nfs][depends] Update libnfs to 6.0.2

### LINKED ISSUE #26103: libnfs API change in most recent release
In the most recent version of libnfs I have had to break the API in order to make READ operations zero-copy.
Zero-copy read will make performance SIGNIFICANTLY better, especially for low-end devices such as RPi and similar.

This change has been planned for a long time but is now ready to go  and will possible land in distros in the next weeks/months/quarters.

A compile-time check is available to determine whether we have the old or new API.

An example on how to make the source code for an applicatiopn compatible with either API si shown below. This is from fio :
```
diff --git a/engines/nfs.c b/engines/nfs.c
index 6bc4af1f..13b55038 100644
--- a/engines/nfs.c
+++ b/engines/nfs.c
@@ -157,16 +157,28 @@ static int queue_write(struct fio_libnfs_options *o, struct io_u *io_u)
 {
        struct nfs_data *nfs_data = io_u->engine_data;
 
+#ifdef LIBNFS_API_V2
+       return nfs_pwrite_async(o->context, nfs_data->nfsfh,
+                               io_u->buf, io_u->buflen, io_u->offset,
+                               nfs_callback, io_u);
+#else
        return nfs_pwrite_async(o->context, nfs_data->nfsfh, io_u->offset,
                                io_u->buflen, io_u->buf, nfs_callback, io_u);
+#endif
 }
 
 static int queue_read(struct fio_libnfs_options *o, struct io_u *io_u)
 {
        struct nfs_data *nfs_data = io_u->engine_data;
 
+#ifdef LIBNFS_API_V2
+       return nfs_pread_async(o->context, nfs_data->nfsfh,
+                               io_u->buf, io_u->buflen, io_u->offset,
+                               nfs_callback, io_u);
+#else
        return nfs_pread_async(o->context, nfs_data->nfsfh, io_u->offset,
                                io_u->buflen, nfs_callback, io_u);
+#endif
 }

```
Kodi will need the same type of ifdef checks.
Sorry for this, but this will make performance a LOT better for especially low-end devices.

### PR DESCRIPTION
**Pull request by sundermann**
## Description

Fix the libnfs API breakage introduced in 6.0.0 and update our depends version.

## Motivation and context

Fix #26103

## How has this been tested?

macOS

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/libnfs/01-MSUWP-compat.patch [hunk 1]
```diff
@@ -1,11 +0,0 @@
---- a/CMakeLists.txt
-+++ b/CMakeLists.txt
-@@ -39,7 +39,7 @@
-   add_definitions(-Dlibnfs_EXPORTS)
- endif()
- 
--if(CMAKE_SYSTEM_NAME STREQUAL Windows)
-+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
```
Comment (enen92) [inline tools/depends/target/libnfs/01-MSUWP-compat.patch:8]: Can't this go upstream @sahlberg? Seems like a non-invasive change

**Comment by sahlberg** [comment]
On Mon, 16 Dec 2024 at 08:15, Miguel Borges de Freitas
***@***.***> wrote:
>
> @enen92 commented on this pull request.
>
> ________________________________
>
> In tools/depends/target/libnfs/01-MSUWP-compat.patch:
>
> > @@ -1,11 +0,0 @@
> ---- a/CMakeLists.txt
> -+++ b/CMakeLists.txt
> -@@ -39,7 +39,7 @@
> -   add_definitions(-Dlibnfs_EXPORTS)
> - endif()
> -
> --if(CMAKE_SYSTEM_NAME STREQUAL Windows)
> -+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
>
> Can't this go upstream @sahlberg? Seems like a non-invasive change

You want something like that in libnfs CMakeList ?
Sure, please send a pull request and I will merge it.

**Comment by sundermann** [comment]
There is already a similar patch upstream https://github.com/sahlberg/libnfs/blob/master/CMakeLists.txt#L93 hence we can remove our patch.

**Comment by fuzzard** [comment]
It was upstreamed by me. Nothing needs to be done, except remove from our version bump as @sundermann has done.

**Comment by enen92** [comment]
Yes sorry was checking on the phone and didn't notice the patch file was actually being removed. Deleted the comment but too late to avoid the ping.
Sorry for the noise.

**Review by thexai — APPROVED** [review]
Tested on Shield

File: xbmc/filesystem/NFSFile.cpp [hunk 2]
```diff
@@ -477,7 +477,11 @@ void CNfsConnection::keepAlive(const std::string& _exportPath, struct nfsfh* _pF
 
   nfs_lseek(pContext, _pFileHandle, 0, SEEK_CUR, &offset);
 
+#ifdef LIBNFS_API_V2
+  int bytes = nfs_read(pContext, _pFileHandle, buffer, sizeof(buffer));
+#else
   int bytes = nfs_read(pContext, _pFileHandle, 32, buffer);
```
Comment (fritsch) [inline xbmc/filesystem/NFSFile.cpp:483]: Even though you did not introduce it - sizeof(buffer) would also work here.

**Comment by sundermann** [comment]
@fuzzard Can you upload `libnfs-6.0.2.tar.gz` to the mirrors?

**Comment by thexai** [comment]
Is already in:
https://mirrors.kodi.tv/build-deps/sources/

**Comment by sundermann** [comment]
Squashed and removed the temporary `BASE_URL` change. Should be good to go once jenkins is done.

**Comment by neo1973** [comment]
@sundermann: Can you please backport the C++ changes to Omega so it can be build on Linux distributions using libnfs 6?

**Comment by sundermann** [comment]
@neo1973 PR submitted

**Pull request merged**